packages/astro/src/core/build/plugins/plugin-content.ts

```diff
@@ -0,0 +1,105 @@
+import type { Plugin as VitePlugin } from 'vite';
+import fsMod from 'node:fs';
+import { addRollupInput } from '../add-rollup-input.js';
+import { type BuildInternals } from '../internal.js';
+import type { AstroBuildPlugin } from '../plugin.js';
+import type { StaticBuildOptions } from '../types.js';
+import { generateContentEntryFile, generateLookupMap } from '../../../content/vite-plugin-content-virtual-mod.js';
+import { joinPaths } from '../../path.js';
+import { fileURLToPath } from 'node:url';
+import type { ContentLookupMap } from '../../../content/utils.js';
+import { CONTENT_RENDER_FLAG } from '../../../content/consts.js';
+import glob from 'fast-glob';
+import { dirname } from 'node:path';
+
+function vitePluginContent(opts: StaticBuildOptions, lookupMap: ContentLookupMap, _internals: BuildInternals): VitePlugin {
+
+	return {
+		name: '@astro/plugin-build-content',
+
+		options(options) {
+			let newOptions = Object.assign({}, options);
+			if (opts.settings.config.output === 'static') {
+				// eslint-disable-next-line @typescript-eslint/no-unused-vars
+				for (const [_collectionName, { type, entries }] of Object.entries(lookupMap)) {
+					let newInputs = Object.values(entries).flatMap(entry => {
+						const input = fileURLToPath(joinPaths(opts.settings.config.root.toString(), entry));
+						const inputs = [`${input}?${collectionTypeToFlag(type)}`];
+						if (type === 'content') {
+							inputs.push(`${input}?${CONTENT_RENDER_FLAG}`)
+						}
+						return inputs;
+					})
+					newOptions = addRollupInput(newOptions, newInputs)
+				}
+			}
+			return newOptions;
+		},
+
+		async generateBundle() {
+			const content = await generateContentEntryFile({ settings: opts.settings, fs: fsMod, lookupMap });
+			this.emitFile({
+				type: 'prebuilt-chunk',
+				code: content,
+				fileName: 'content/index.mjs'
+			})
+		},
+
+		async writeBundle(options, bundle) {
+			const dist = options.dir!
+			const cache = dist.replace('dist', 'node_modules/.astro/content-cache')
```

Comment:
Using `node_modules/.astro/content-cache` since the asset generation uses this directory as well, but this caching logic should be centralized!

Reply:
Just a heads up that we have a `config.cacheDir`: https://docs.astro.build/en/reference/configuration-reference/#cachedir

Reply:
Ahh excellent, thank you!